src/listpack.c

```diff
@@ -933,27 +939,35 @@ unsigned char *lpPrepend(unsigned char *lp, unsigned char *s, uint32_t slen) {
 unsigned char *lpAppend(unsigned char *lp, unsigned char *ele, uint32_t size) {
     uint64_t listpack_bytes = lpGetTotalBytes(lp);
     unsigned char *eofptr = lp + listpack_bytes - 1;
-    return lpInsertString(lp,ele,size,eofptr,LP_BEFORE,NULL);
+    return lpInsert(lp,ele,NULL,size,eofptr,LP_BEFORE,NULL);
 }
 
 /* Append the specified integer element 'lval' at the end of the listpack. */
-unsigned char *lpAppendInteger(unsigned char *lp, int64_t lval) {
+unsigned char *lpAppendInteger(unsigned char *lp, long long lval) {
     uint64_t listpack_bytes = lpGetTotalBytes(lp);
     unsigned char *eofptr = lp + listpack_bytes - 1;
-    return lpInsertInteger(lp,lval,eofptr,LP_BEFORE,NULL);
+    return lpInsertInteger(lp, lval, eofptr, LP_BEFORE, NULL);
 }
 
 /* Remove the element pointed by 'p'. */
-unsigned char *lpReplace(unsigned char *lp, unsigned char *p, unsigned char *s, uint32_t slen) {
-    return lpInsertString(lp, s, slen, p, LP_REPLACE, NULL);
+unsigned char *lpReplace(unsigned char *lp, unsigned char **p, unsigned char *s, uint32_t slen) {
+    return lpInsert(lp, s, NULL, slen, *p, LP_REPLACE, p);
+}
```

Comment:
why did you change lpReplace to take a double pointer (`**`) `p`?
I suppose just for symmetry with lpReplaceInteger?
let's at least document what it does with it (copy the text from lpReplaceInteger?).

Reply:
Yes, just symmetry with `lpReplaceInteger`.